Automattic/sensei
# Add course theme templates

Fixes #4368

### Changes proposed in this Pull Request

* This PR introduces the new Course Theme templates, and makes the lessons and quizzes use them when the Sensei Theme is selected.
  * The approach used here is to load a cleaner template as wrapper (similar to what we do for unsupported themes loading the `page.php` template). Then we load the respective template through the `the_content` filter, rendering the blocks we have there.
  * The blocks rendering in the template is a similar approach to what will be used for FSE. [More details here](https://github.com/WordPress/gutenberg/blob/trunk/lib/full-site-editing/template-loader.php#L176).
  * The purpose of the used approach is to be more ready to support FSE in the future, so our main elements will be implemented as blocks.
* It also fixes some versions in the PHPDoc, because the code will be released in `3.13.4`, even behind a feature flag.

### Testing instructions


* Enable the feature flag: `add_filter( 'sensei_feature_flag_course_theme', '__return_true' );`
* Create a course with lessons and quizzes.
* Add some blocks to the lessons to make sure they were rendered properly.
* Make sure the Lessons are using the default theme in the frontend.
* Select the "Sensei Theme" as the "Course Theme" (sidebar).
* Make sure the Lesson is loading the Sensei theme in the frontend.
* Make sure you can override the templates through your theme.
  * Copy the folder `/templates/course-theme` to `/yourthemefolder/sensei/`.
  * Make some changes to that files.
  * Make sure it reflects in the frontend.


### Screenshot / Video

Note that it's still pretty rough. It'll be improved as we create the new blocks and add the styles.

<img width="1409" alt="Screen Shot 2021-11-08 at 11 30 36" src="https://user-images.githubusercontent.com/876340/140760311-0e903905-4073-4e8e-ba1f-333cbe36436f.png">

<img width="1411" alt="Screen Shot 2021-11-08 at 11 30 57" src="https://user-images.githubusercontent.com/876340/140760356-55327435-8cd3-4342-84d7-d364464f1696.png">

<img width="1408" alt="Screen Shot 2021-11-08 at 11 35 12" src="https://user-images.githubusercontent.com/876340/140760946-307aa3a5-b5f9-4db9-8da8-b6e3297a26e9.png">
